### ISSUE TITLE
Per-category thresholds in the CLI exit-code contract

### ISSUE THREAD

**Issue by Jamie-BitFlight** [Jamie-BitFlight/textlint-rule-preset-ste-ai#17]
## Observed

The exit-code contract, asserted by `scripts/ci/check-exit-codes.sh`:

```
deterministic-only run produced 234 diagnostics across 18 files, exit code 1 as documented
clean document exited 0 as documented
usage error exited 2 as documented
```

Exit 1 on any finding. `--min-severity` filters which findings count, and `--fail-on-review` controls whether `review-required` participates. There is no count threshold.

## The gap

For a rule with a known false-positive rate, "fail on the first hit" and "never fail" are the only settings, and neither is useful as a gate. A team adopting this on an existing documentation set has no way to say "fail if this file has more than three of these", which is the setting that lets a gate be adopted incrementally without a mass rewrite first.

## Prior art

humanizer-stack's scanners take this shape: `--strict` fails only when a category reaches a per-category count, defaulting to 2 rather than 1, on the stated grounds that these are heuristics with known false positives. Noted in `docs/extension-roadmap.md`.

## Proposal

Per-category and per-rule count thresholds in the shared configuration, consumed by the CLI when deciding its exit code:

```json
{ "gate": { "maxPerFile": { "deterministic-violation": 0, "review-required": 3 } } }
```

The decision belongs in the CLI, not in the core. The core reports what it found; the exit code is a policy about that report. `src/cli/main.ts` already owns the exit-code contract and `UsageError`.

## Related

#10 (per-instance suppression) is the other half of making findings manageable. Thresholds handle aggregate noise; suppression handles a specific known-good usage. Neither substitutes for the other.
